swap/swap.go

```diff
@@ -314,22 +314,52 @@ func createOwner(prvkey *ecdsa.PrivateKey) *Owner {
 	}
 }
 
-// Add is the (sole) accounting function
-// Swap implements the protocols.Balance interface
-func (s *Swap) Add(amount int64, peer *protocols.Peer) (err error) {
+// checkBalance checks that the amount would not result in crossing the disconnection threshold
+func (s *Swap) checkBalance(amount int64, peer *protocols.Peer) (err error) {
 	swapPeer := s.getPeer(peer.ID())
 	if swapPeer == nil {
 		return fmt.Errorf("peer %s not a swap enabled peer", peer.ID().String())
 	}
-	swapPeer.lock.Lock()
-	defer swapPeer.lock.Unlock()
 
 	// check if balance with peer is over the disconnect threshold and if the message would increase the existing debt
 	balance := swapPeer.getBalance()
 	if balance >= s.params.DisconnectThreshold && amount > 0 {
```

Comment:
This change was introduced by @mortelli with the idea to always allow debt-reducing messages. Let him comment on this; in any case, if this would have to change, it is an important change and then it should be done in a separate PR

Reply:
I vaguely remember this... This comment is not a blocker for me, but would like to get reassurance from @mortelli :)

Reply:
indeed, i introduced this change through PR #1922. 

the only difference in checking `amount` instead of `balance + amount` is **one** message.

in the first case, one message will go _over_ the threshold and then stop the flow; in the second, the threshold will never be reached or surpassed, but there might be a bigger difference between the threshold and the final balance before subsequent messages are stopped.

while correct, the situation you mentioned is unlikely to come up: if a message is priced at `1000`, we should never have set a threshold at `500`.

assuming then, that `amount` is comparatively small in regards to the threshold, the impact of having this little bit of extra slack should be minimal.